server/controllers/notifications.js

```diff
@@ -2,7 +2,7 @@ const asyncHandler = require('express-async-handler');
 const Notifications = require('../models/Notifications');
 
 exports.getAllNotifications = asyncHandler(async(req, res) => {
-    const receiverProfileId = req.body.profileId;
+    const receiverProfileId = req.params.id;
```

Comment:
You can maybe refactor controllers to get notifications with the user's id (request object).
Eg: `req.user.id`